arcaflow_plugin_rtla/rtla_plugin.py

```diff
@@ -181,7 +213,67 @@ def run_timerlat(
                 total_usr_latency[label] = line_list[3]
 
         # Provide the rtla command formatted data as debug output
-        print(output)
+        print(timerlat_output)
+
+        # Example time series output from the tracer (truncated)
+        # <idle>-0      [009] d. 123.769498: #1002 context    irq timer_latency  458 ns
+        #  <...>-625890 [009] .. 123.769499: #1002 context thread timer_latency 666 ns
+        #  <...>-625890 [009] .. 123.769500: #1002 context user-ret timer_latency 499 ns
+        # <idle>-0      [002] d. 123.769528: #1003 context    irq timer_latency  587 ns
+        #  <...>-625883 [002] .. 123.769532: #1003 context thread timer_latency 712 ns
+        #  <...>-625883 [002] .. 123.769534: #1003 context user-ret timer_latency 462 ns
+
+        if params.enable_time_series and trace_file_exists:
+            timeseries_lines = iter(timeseries_output.splitlines())
+
+            # The calculation of the offset from uptime to current time certainly means
+            # that our time series is not 100% accurately aligned to the nanosecond, but
+            # the relative times will be accurate. We'll accept this as good enough.
+            uptime_offset = time.time() - time.monotonic()
+
+            if all(False for _ in timeseries_lines):
+                print(
+                    "No results reading tracer output; "
+                    "Skipping time series collection\n"
+                )
+
+            for line in timeseries_lines:
+                line_list = line.split()
+                # Because the tracer format is dependent on the underlying OS and cannot
+                # be controlled by the container, check the tracer output format and
+                # break gracefully if we don't recognize it
+                try:
+                    cpu = int(line_list[1][1:-1])
+                    uptimestamp = float(line_list[3][:-1])
+                    timestamp = str(
+                        datetime.fromtimestamp(uptimestamp + uptime_offset)
+                        .astimezone()
+                        .isoformat()
+                    )
+                    context = str(line_list[6])
+                    latency = int(line_list[8])
```

Comment:
Consider using a regex with capture groups, instead.  I think it will ultimately be simpler (to read and maintain, at least), and the "unknown format" result will be much simpler and more obvious.